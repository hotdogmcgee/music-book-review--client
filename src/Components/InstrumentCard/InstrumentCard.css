.InstrumentCard {
  height: -webkit-fill-available;
}

.img-wrap {
  height: 67%;
}
.InstrumentCard img {
  width: 100%;
  height: -webkit-fill-available;
  display: block;
}

.InstrumentCard h2 {
  margin: 0;
}

.InstrumentCard__title {
  color: white;
  text-shadow: 2px 2px black;
}

.InstrumentCard a {
  text-decoration: none;
}
