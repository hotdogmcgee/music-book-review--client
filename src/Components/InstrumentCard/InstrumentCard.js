import React from "react";
import { Link } from "react-router-dom";
import "./InstrumentCard.css";

export default function InstrumentCard(props) {
  const { title } = props;
  let imageLink = "";

  if (title) {
    switch (title) {
      case "guitar":
        imageLink = process.env.PUBLIC_URL + "/Photos/guitar-cropped.jpg";
        break;

      case "piano":
        imageLink = process.env.PUBLIC_URL + "/Photos/piano-resize.jpg";
        break;

        case 'clarinet':
          imageLink = process.env.PUBLIC_URL + "/Photos/clarinet.jpg";
          break;

          case 'violin':

          imageLink = process.env.PUBLIC_URL + "/Photos/violin.jpg";
          break;

      default:
        imageLink = "";
    }
  }

  return (
    <div className={`InstrumentCard`}>
      <Link to={`/category/instrument/${title}`}>
        <div className="img-wrap">
          {/* {imageLink.length > 0 ? <img src={imageLink} alt={title} /> : ""} */}
          <img src={imageLink} alt={title} />
        </div>

        <div className="InstrumentCard__title">
          <h2 className="capitalize">{title}</h2>
        </div>
      </Link>
    </div>
  );
}
