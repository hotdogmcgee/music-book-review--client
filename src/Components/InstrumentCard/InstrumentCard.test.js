import React from "react";
import InstrumentCard from "./InstrumentCard";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe.only("InstrumentCard component", () => {
  it("renders a InstrumentCard by default", () => {
    const wrapper = shallow(<InstrumentCard title='guitar' />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});