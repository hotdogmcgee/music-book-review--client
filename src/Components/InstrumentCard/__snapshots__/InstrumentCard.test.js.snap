// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentCard component renders a InstrumentCard by default 1`] = `
<div
  className="InstrumentCard"
>
  <Link
    to="/category/instrument/guitar"
  >
    <div
      className="img-wrap"
    >
      <img
        alt="guitar"
        src="guitar.jpg"
      />
    </div>
    <div
      className="InstrumentCard__title"
    >
      <h2
        className="capitalize"
      >
        guitar
      </h2>
    </div>
  </Link>
</div>
`;
